ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/multiplechoice/SelectMultipleOptionAdapter.kt

```diff
@@ -42,26 +42,25 @@ class SelectMultipleOptionAdapter(
     holder.bind(options[position])
 
     holder.binding.checkbox.isChecked = position in selectedPositions
+    holder.binding.checkbox.setOnCheckedChangeListener { _, _ ->
+      handleItemStateChanged(holder.adapterPosition)
+    }
   }
 
-  override fun getItemCount(): Int = options.size
-
-  override fun getItemId(position: Int): Long = position.toLong()
-
-  override fun getPosition(key: Long): Int = key.toInt()
-
-  override fun handleItemStateChanged(position: Int, selected: Boolean) {
+  private fun handleItemStateChanged(position: Int) {
     if (position in selectedPositions) {
       selectedPositions.remove(position)
     } else {
       selectedPositions.add(position)
     }
 
-    viewModel.updateResponse(
+    handleOptionSelected.invoke(
```

Comment:
Should be able to just call it directly instead of using `invoke`: `handleOptionSelected(options.filterIndexed {}...)`